application/src/main/java/org/thingsboard/server/controller/plugin/TbWebSocketHandler.java

```diff
@@ -120,6 +134,9 @@ public void afterConnectionEstablished(WebSocketSession session) throws Exceptio
                 return;
             }
             internalSessionMap.put(internalSessionId, new SessionMetaData(session, sessionRef, maxMsgQueuePerSession));
+
+            pingExecutor.scheduleWithFixedDelay(() -> internalSessionMap.get(internalSessionId).sendPing(), 10000, 10000, TimeUnit.MILLISECONDS);
```

Comment:
No need to schedule separate function execution per session. You should execute this function and loop through all the sessions. This code also has a problem that you do not cancel this task after session is closed. Do you wait/analyze the reply or not really? If you wait for the reply, it may be a very time-consuming task (better to send ping in the same thread pool where you send the updates over websockets if this call is blocking).